common/src/main/java/io/pravega/common/util/Retry.java

```diff
@@ -206,7 +210,39 @@ private RetryAndThrowBase(Class<ThrowsT> throwType, RetryWithBackoff params) {
             }
             throw new RetriesExhaustedException(last);
         }
-
+        
+        public CompletableFuture<Void> runInExecutor(final Runnable task,
+                                                     final ScheduledExecutorService executorService) {
+            Preconditions.checkNotNull(task);
+            CompletableFuture<Void> result = new CompletableFuture<>();
```

Comment:
You don't need this `result`. Just return the thing you get back from `FutureHelpers.loop`. Maybe use an `AtomicBoolean canLoop` to figure out when you're done. If you have an exception, you can just throw it out of the `exceptionally` block  - the loop will catch that and return the appropriate result.